lib/src/workspace.rs

```diff
@@ -140,6 +142,17 @@ impl Workspace {
         })
     }
 
+    pub fn create_workspace_root(workspace_root: &Path) -> std::io::Result<PathBuf> {
+        match fs::create_dir(workspace_root) {
+            Ok(()) => {}
+            Err(_) if workspace_root.is_dir() => {}
+            Err(e) => return Err(e),
+        }
+
```

Comment:
If we don't mind creating intermediate directories, this is basically `std::fs::create_dir_all()`.

If not, maybe we can add this function to `file_util`? The functionality is generic and has little relation to Workspace.

Reply:
Split this out to #2829